int32 left
int32 right
int32 top
int32 bottom
float32 center_x
float32 center_y
float32 center_z